src/core/modal.slice.ts

```diff
@@ -0,0 +1,28 @@
+import { createSlice } from "@reduxjs/toolkit";
+
```

Comment:
Really a detail but since you use the `modalId`several times you could consider defining it as a type:
```ts
type ModalId = string;

interface PayloadProps {
  payload: {
    modalId: ModalId;
  };
}
interface StateProps {
  modalIds: ModalId[];
}

Reply:
And then export it and use it in other files..?